src/Company/ui/JobListings.tsx

```diff
@@ -7,33 +7,32 @@ import { CompanyPositions } from "../CompanyPositions";
 import { ApplyToJobButton } from "./ApplyToJobButton";
 import { Player } from "@player";
 
-interface IJobListingsProps {
+interface JobListingsProps {
   company: Company;
   currentPosition: CompanyPosition | null;
 }
 
-export function JobListings(props: IJobListingsProps): React.ReactElement {
-  const { company, currentPosition } = props;
-
+export function JobListings({ company, currentPosition }: JobListingsProps): React.ReactElement {
+  const jobNames = [...company.companyPositions];
+  const qualifiedJobs = new Set(jobNames.filter((jobName) => Player.isQualified(company, CompanyPositions[jobName])));
   const jobsToShow = [];
-  for (const jobName of company.companyPositions) {
-    const offeredPos = CompanyPositions[jobName];
-    const underqualified = !Player.isQualified(props.company, offeredPos);
-    const isCurrentPosition = jobName == props.currentPosition?.name;
-    const nextPos = offeredPos.nextPosition && CompanyPositions[offeredPos.nextPosition];
-    const overqualified = nextPos != null && Player.isQualified(props.company, nextPos);
-    const shouldShowApplyButton =
-      !isCurrentPosition && !overqualified && (!underqualified || offeredPos.requiredReputation == 0);
-
-    if (shouldShowApplyButton) {
-      jobsToShow.push(offeredPos);
+  for (const jobName of jobNames) {
+    if (jobName === currentPosition?.name) continue;
+    const job = CompanyPositions[jobName];
+    const nextJobName = job.nextPosition;
+    // Don't show a job if we already qualify for a later job
```

Comment:
Clarify that "overqualified" jobs are company-specific
```suggestion
    // Don't show a job if we already qualify for a later job offered by this company
```